### PR TITLE
NAS-116577 / 23.10 / NAS-116577: Improve Handled unknown error in sentry

### PR DESCRIPTION
**Pull request by AlexKarpov98**
Updated one package to avoid this error: **[(...).transition is not a function]**

Found a way to extend the default error handler, 
and will return **JSON.stringify(error)** if no matches found for error handler to avoid **"Handled unknown error. "**

### CODE REVIEW

**Review by undsoft — CHANGES_REQUESTED** [review]
1. Second argument in `extractor` is the `defaultExtractor`. You call the default extractor and then if it fails, do something else.
2. It would be good to try to extract error message, not just put in JSON, as it may be huge with stacktrace and other data.
See `EntityUtils.handleWsError` and `EntityUtils.errorReport`.

**Comment by AlexKarpov98** [comment]
would be good to check how much new default messages will we get 
`iXsystems [sentry] default handled error`

**Review by undsoft — CHANGES_REQUESTED** [review]
`error.message` is already extracted by default extractor. Not sure what `error.name` is.

You need to use custom extractor to extract error messages that we may get from middleware. You can look at `WebsocketError` and `Job` for relevant fields and at `EntityUtils` for how errors are currently being processed.

**Comment by undsoft** [comment]
Bump.

File: src/app/helpers/sentry-custom-extractor.helper.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  errorCandidate: WebsocketError | Job | Error | any,
+  defaultExtractor: (error: unknown) => unknown,
+): unknown => {
+  const defaultExtractorResults = defaultExtractor(errorCandidate);
+
+  if (defaultExtractorResults) {
+    return defaultExtractorResults;
+  }
+
+  const errorReportValue = errorReport(errorCandidate as WebsocketError);
```
Comment (undsoft) [inline src/app/helpers/sentry-custom-extractor.helper.ts:25]: `as WebsocketError | Job`

File: src/app/helpers/sentry-custom-extractor.helper.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import ErrorStackParser from 'error-stack-parser';
+import { Job } from 'app/interfaces/job.interface';
+import { WebsocketError } from 'app/interfaces/websocket-error.interface';
+
+const errorReport = (err: WebsocketError | Job): string | void => {
```
Comment (undsoft) [inline src/app/helpers/sentry-custom-extractor.helper.ts:5]: Isn't really `errorReport`, morel like `extractApiError`

**Review by undsoft — APPROVED** [review]
Looks good.

**Comment by undsoft** [comment]
Would be good to fix comments as well. @AlexKarpov98

**Comment by AlexKarpov98** [comment]
@undsoft, yes, sorry, auto push triggered after approval.
Changes are here now: https://github.com/truenas/webui/pull/7403

**Pull request merged**